upload-api/config.js

```diff
@@ -6,20 +6,28 @@
  * see: https://docs.sst.dev/advanced/testing#how-sst-bind-works
  */
 import * as ed25519 from '@ucanto/principal/ed25519'
-import { DID } from '@ucanto/validator'
+import * as DID from '@ipld/dag-ucan/did'
 
 /**
  * Given a config, return a ucanto Signer object representing the service
  *
  * @param {object} config
- * @param {string} [config.UPLOAD_API_DID] - public identifier of the running service. e.g. a did:key or a did:web
  * @param {string} config.PRIVATE_KEY - multiformats private key of primary signing key
  */
  export function getServiceSigner(config) {
   const signer = ed25519.parse(config.PRIVATE_KEY)
-  const did = config.UPLOAD_API_DID
-  if (!did) {
-    return signer
+  return signer
+}
+
+/**
+ * Given a config, return a ucanto principal
+ *
+ * @param {{ UPLOAD_API_DID: string } | { PRIVATE_KEY: string }} config
+ * @returns {import('@ucanto/interface').Principal}
+ */
+export function getServerPrincipal(config) {
```

Comment:
We've got `service` and `server` and i'm not clear if there is a difference.

should this be
```suggestion
export function getServicePrincipal(config) {
```

Reply:
great question. It seems to me that 'service' gets used frequently for the object with a few methods, one per ucan capability name. But then you pass that 'service' along with some other options (like `id: Principal`) to the ucanto `Server` constructor to make a 'ucanto server' that exposes the (js obj) 'service'. That's my thinking.